hooks/tk-multi-publish2/publish_file.py

```diff
@@ -0,0 +1,265 @@
+                    "default": "default_value",
+                    "description": "One line description of the setting"
+            }
+        The type string should be one of the data types that toolkit accepts
+        as part of its environment configuration.
+        """
+        return {
+            "File Types": {
+                "type": "list",
+                "default": [
+                    ["Alias", "wire"],
+                    ["Alembic Cache", "abc"],
+                    ["3dsmax Scene", "max"],
+                    ["NukeStudio Project", "hrox"],
+                    ["Houdini Scene", "hip", "hipnc"],
+                    ["Maya Scene", "ma", "mb"],
+                    ["FBX Translation", "fbx"],
+                    ["Nuke Script", "nk"],
+                    ["Photoshop Image", "psd", "psb"],
+                    ["VRED", "vpb", "vpe"],
+                    ["Rendered Image", "dpx", "exr"],
+                    ["Texture", "tiff", "tx", "tga", "dds"],
+                    ["Image", "jpeg", "jpg", "png"],
+                    ["Movie", "mov", "mp4"],
+                    ["PDF", "pdf"],
+                    ["IGS Translation", "igs"],
+                    ["JT Translation", "jt"],
+                    ["CATPart Translation", "CATPart", "catpart"],
+                    ["STP Translation", "stp"],
+                    ["STL Translation", "stl"],
+                    ["WREF Translation", "wref"],
+                    ["OSB", "osb"],
+                    ["Office File", "doc", "docx", "xls", "xlsx", "ppt", "pptx"]
+                ],
+                "description": (
+                    "List of file types to include. Each entry in the list "
+                    "is a list in which the first entry is the Shotgun "
+                    "published file type and subsequent entries are file "
+                    "extensions that should be associated."
+                )
+            },
+        }
+
+    def get_publish_fields(self, settings, item):
+        """
+        Get additional fields that should be used for the publish. This
+        dictionary is passed on to :meth:`tank.util.register_publish` as the
+        ``sg_fields`` keyword argument.
+        If publish_fields is not defined as a ``property`` or
+        ``local_property``, this method will return an empty dictionary.
+        :param settings: This plugin instance's configured settings
+        :param item: The item to determine the publish template for
+        :return: A dictionary of field names and values for those fields.
+        """
+        if item.parent.properties.get("sg_status_list") is not None:
+            return {"sg_status_list": item.parent.properties["sg_status_list"]}
+        return item.get_property("publish_fields", default_value={})
+
+        ############################################################################
+        # protected methods
+
+    def _copy_work_to_publish(self, settings, item):
```

Comment:
not sure this method is useful in this plugin, you should inherit from the default one
it's an heritage from Renault where the copy method was not working because of the Windows permissions